locks.go

```diff
@@ -143,3 +148,31 @@ func (l *LockRedis) Unlock(ctx context.Context) error {
 	}
 	return nil
 }
+
+// LockMemcached is a wrapper around github.com/alessandro-c/gomemcached-lock that implements the DistLocker interface.
+type LockMemcached struct {
+	locker    *lock.Locker
+	mutexName string
+}
+
+// NewLockMemcached creates a new instance of LockMemcached.
+func NewLockMemcached(client *memcache.Client, mutexName string) *LockMemcached {
+	adapter := gomemcache.New(client)
+	locker := lock.New(adapter, mutexName + ":Lock", "")
+	return &LockMemcached{
+		locker:    locker,
+		mutexName: mutexName,
+	}
+}
+
+// Lock locks the lock in Memcached.
+func (l *LockMemcached) Lock(ctx context.Context) error {
+	o := func() error { return l.locker.Lock(time.Minute) }
+	b := backoff.WithMaxRetries(backoff.NewConstantBackOff(10*time.Millisecond), 1000)
+	return backoff.Retry(o, b)
```

Comment:
Why is retry needed to acquire the lock?

Reply:
The underlying lock of `LockRedis` is a blocking lock, as it retries internally till it gets the lock or expires.
https://github.com/go-redsync/redsync/blob/master/mutex.go#L75

The underlying lock of `LockMemcached` is a non-blocking lock, as it returns error immediately if it is locked already.
https://github.com/alessandro-c/gomemcached-lock/blob/master/locker.go#L47

In that case we have to retry in LockMemcached. Using `backoff` is just a convenient way so we can easily switch from constant backoff to exponential off, do it with timeout context or with max retries, etc.

Reply:
Hm... i see now why you need the retry.
The retry mechanism should be configurable in this case, it should be a part of the `LockMemcached` struct with some sane defaults.

Reply:
Okay, that makes sense. I'll do that.